### PR TITLE
Consider environment when configuring k8s clients

### PR DESCRIPTION
**Pull request by cbandy**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

OLM scanners expect to be able to redirect Kubernetes API calls by setting environment variables. We use the most basic kube client configuration functions which don't consider env vars.

**What is the new behavior (if this is a feature change)?**

The higher-level configuration functions of k8s.io/client-go check `KUBECONFIG` env var and `~/.kube/config` directory and fall back to in-cluster config, if available.

**Other information**:

Issue: [ch7130]

### CODE REVIEW

File: kubeapi/client_config.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Match the settings applied by sigs.k8s.io/controller-runtime@v0.4.0;
+	// see https://github.com/kubernetes-sigs/controller-runtime/issues/365.
+	if config.QPS == 0.0 {
+		config.QPS = 20.0
+		config.Burst = 30.0
+	}
```
Comment (cbandy) [inline kubeapi/client_config.go]: I came across this logic in the referenced package, but I'm not _certain_ we should change them at this time. [The defaults](https://godoc.org/k8s.io/client-go/rest#Config) are `QPS: 5, Burst: 10`. Thoughts?

Comment (cbandy) [inline kubeapi/client_config.go]: I came across this logic in the referenced package, but I'm not _certain_ we should change them at this time. [The defaults](https://godoc.org/k8s.io/client-go/rest#Config) are `QPS: 5, Burst: 10`. Thoughts?

Comment (jkatz) [reply kubeapi/client_config.go]: @cbandy My preference is if it ain't broke, don't fix it.

Comment (jkatz) [reply kubeapi/client_config.go]: Which is another way of saying that if this fixes something that is broke, sure...

Comment (jkatz) [reply kubeapi/client_config.go]: Let's leave it.

File: kubeapi/client_config.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	loader.ExplicitPath = kubeconfig
+
+	// The deferred loader tries an in-cluster config if the default loading
+	// rules produce no results.
+	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
+		loader, &clientcmd.ConfigOverrides{},
+	).ClientConfig()
+}
+
+func NewClientConsideringFlag(kubeconfig string) (*rest.Config, *kubernetes.Clientset, error) {
```
Comment (jkatz) [inline kubeapi/client_config.go]: Pleaes add documentation on this function

File: kubeapi/client_config.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	config, err := loadClientConfig(kubeconfig)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	return config, clientset, err
+}
+
+func NewControllerClientConsideringFlag(kubeconfig string) (*rest.Config, *kubernetes.Clientset, error) {
```
Comment (jkatz) [inline kubeapi/client_config.go]: Please add documentation on this function

**Pull request merged**